Doc/library/sqlite3.rst

```diff
@@ -2358,6 +2340,111 @@ can be found in the `SQLite URI documentation`_.
 .. _SQLite URI documentation: https://www.sqlite.org/uri.html
 
 
+.. _sqlite3-howto-row-factory:
+
+How to create and use row factories
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+By default, :mod:`!sqlite3` represent fetched rows as :class:`tuples <tuple>`.
+If a :class:`!tuple` does not suit your needs,
+use the :class:`sqlite3.Row` class or a custom :attr:`~Connection.row_factory`.
+
+:class:`!Row` provides indexed and case-insensitive named access to columns,
+with low memory overhead and minimal performance impact.
+In order to use :class:`Row` as a row factory,
+simply assign it to the :attr:`Connection.row_factory` attribute:
+
+.. doctest::
+
+   >>> con = sqlite3.connect(":memory:")
+   >>> con.row_factory = sqlite3.Row
+
+Query results are now processed using :class:`Row`:
+
+.. doctest::
+
+   >>> res = con.execute("SELECT 'Earth' AS name, 6378 AS radius")
+   >>> row = res.fetchone()
+   >>> row.keys()
+   ['name', 'radius']
+   >>> row[0]  # Access by index.
+   'Earth'
+   >>> row["name"]    # Access by name.
+   'Earth'
+   >>> row["RADIUS"]  # Column names are case-insensitive.
+   6378
+
+It is also possible to assign row factories to cursors using
+:attr:`Cursor.row_factory`:
+
+.. doctest::
+
+   >>> cur = con.cursor()
+   >>> cur.row_factory == con.row_factory
+   True
+   >>> cur.row_factory = None  # Override cursor row factory.
```

Comment:
At least initially, I found it a bit hard to follow here that the cursor row factory was being set to `None` and then had the default behavior, at least without more explicit comments.

Also, it might make more sense to move this example to after the example of the custom connection factory, to avoid splitting up the two different things you can set `row_factory` to with an example that's quite different. You could also use that to provide an example of a `row_factory` that is itself a class rather than a function.

Reply:
I'll rework this example completely. Let's see how it turns out.

Reply:
I decided to ditch this example after rewriting the row factory references. If you still think this example is needed, we may consider adding a minimal version to the reference instead.